x-pack/plugins/licensing/server/plugin.ts

```diff
@@ -4,145 +4,142 @@
+      refresh,
+      license$,
+    };
   }
 
-  private async fetchInfo(core: CoreSetup, clusterSource: string, pollingFrequency: number) {
-    this.logger.debug(
-      `Calling [${clusterSource}] Elasticsearch _xpack API. Polling frequency: ${pollingFrequency}`
+  private createLicensePoller(clusterClient: IClusterClient, pollingFrequency: number) {
+    const manualRefresh$ = new Subject();
+    const intervalRefresh$ = timer(0, pollingFrequency);
+    const refresh$ = merge(intervalRefresh$, manualRefresh$).pipe(takeUntil(this.stop$));
+
+    const { update$, fetchSubscription } = createLicenseUpdate(refresh$, () =>
+      this.fetchLicense(clusterClient)
     );
 
-    const cluster = await core.elasticsearch.dataClient$.pipe(first()).toPromise();
+    this.licenseFetchSubscription = fetchSubscription;
+
+    const license$ = update$.pipe(
+      tap(license => {
+        this.logger.debug(
+          'Imported license information from Elasticsearch:' +
+            [
+              `type: ${license.type}`,
+              `status: ${license.status}`,
+              `expiry date: ${moment(license.expiryDateInMillis, 'x').format()}`,
+            ].join(' | ')
+        );
+      })
+    );
+
+    return {
+      refresh: () => {
+        this.logger.debug('Requesting Elasticsearch licensing API');
+        manualRefresh$.next();
+      },
+      license$,
+    };
+  }
 
+  private fetchLicense = async (clusterClient: IClusterClient): Promise<ILicense> => {
     try {
-      const response = await cluster.callAsInternalUser('transport.request', {
+      const response = await clusterClient.callAsInternalUser('transport.request', {
         method: 'GET',
         path: '/_xpack',
       });
-      const rawLicense = response && response.license;
-      const features = (response && response.features) || {};
-      const licenseInfoChanged = this.hasLicenseInfoChanged(rawLicense);
-
-      if (!licenseInfoChanged) {
-        return { license: false, error: null, features: null };
-      }
-
-      const currentLicense = this.poller.subject$.getValue();
-      const licenseInfo = [
-        'type' in rawLicense && `type: ${rawLicense.type}`,
-        'status' in rawLicense && `status: ${rawLicense.status}`,
-        'expiry_date_in_millis' in rawLicense &&
-          `expiry date: ${moment(rawLicense.expiry_date_in_millis, 'x').format()}`,
-      ]
-        .filter(Boolean)
-        .join(' | ');
-
-      this.logger.info(
-        `Imported ${currentLicense ? 'changed ' : ''}license information` +
-          ` from Elasticsearch for the [${clusterSource}] cluster: ${licenseInfo}`
-      );
 
-      return { license: rawLicense, error: null, features };
-    } catch (err) {
+      const normalizedLicense = normalizeServerLicense(response.license);
+      const signature = sign({ license: normalizedLicense, features: response.features });
+
+      return new License({
+        license: normalizedLicense,
+        features: response.features,
```

Comment:
should we normalize features? we have the next logic 
on the client: https://github.com/elastic/kibana/blob/60c596cb3bab338a259ad7246b1d3065eaad89ca/x-pack/legacy/plugins/xpack_main/public/services/xpack_info.js#L30